server/src/main/java/io/spine/server/entity/storage/PersistenceInfo.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity.storage;
+
+import io.spine.annotation.Internal;
+import io.spine.server.entity.storage.enumeration.EnumPersistenceTypes;
+import io.spine.server.entity.storage.enumeration.EnumType;
+import io.spine.server.entity.storage.enumeration.Enumerated;
+
+import java.io.Serializable;
+import java.lang.reflect.Method;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.server.entity.storage.enumeration.EnumConverters.forType;
+import static io.spine.server.entity.storage.enumeration.EnumType.ORDINAL;
+
+/**
+ * An info about how the {@link EntityColumn} is persisted in the data storage.
+ *
+ * <p>This class provides user with the info about {@link EntityColumn} persistence type as well
+ * as the conversion method between the entity column values and the persistence values.
+ *
+ * @author Dmytro Kuzmin
+ * @see Column
+ */
+@Internal
+final class PersistenceInfo implements Serializable {
```

Comment:
The name is way too generic. 

Please think of this class as of something performing an action, not just as of storage for some ephemeral information (as it is now). Then you'll be able to give it some responsibility and name it properly.